infrastructure/src/main/java/org/corfudb/infrastructure/orchestrator/workflows/HealNodeWorkflow.java

```diff
@@ -89,4 +92,43 @@ public void impl(@Nonnull CorfuRuntime runtime) throws Exception {
             newLayout = new Layout(runtime.getLayoutView().getLayout());
         }
     }
+
+    /**
+     * This action attempts to restore redundancy for all servers across all segments
+     * starting from the oldest segment. It then collapses the segments once the set of
+     * servers in the 2 oldest subsequent segments are equal.
+     */
+    class RestoreRedundancyAndMergeSegments extends Action {
+        @Nonnull
+        @Override
+        public String getName() {
+            return "RestoreRedundancyAndMergeSegments";
+        }
+
+        @Override
+        public void impl(@Nonnull CorfuRuntime runtime) throws Exception {
+            // Catchup all servers across all segments.
+            while (newLayout.getSegments().size() > 1) {
+
+                // Currently the state is transferred for the complete segment.
+                // TODO: Add stripe specific transfer granularity for optimization.
```

Comment:
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Complete the task associated to this TODO comment. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1135)